lib/Cake/Controller/Controller.php

```diff
@@ -389,15 +396,19 @@ public function __isset($name) {
  * @return mixed The requested value for valid variables/aliases else null
  */
 	public function __get($name) {
+		$message = "Property $name is deprecated. Use CakeRequest::$name instead.";
 		switch ($name) {
 			case 'base':
 			case 'here':
 			case 'webroot':
 			case 'data':
+				trigger_error($message, E_USER_DEPRECATED);
```

Comment:
New deprecation error messages shouldn't be added in bugfix releases.

Reply:
Would it be okay to write to a log file instead?